### PR TITLE
Add new model fields

### PR DESCRIPTION
**Pull request by nishanthvijayan**
- Added name in Transition
- Added allowed_groups in State & remove allowed_groups from Transition
- Added allowed_groups field to Workflow.
- Added checks to verify whether user is in the allowed_groups of a Workflow before allowing create. + Tests for the same

Solves #41, #44 , #51

### CODE REVIEW

**Comment by nishanthvijayan** [comment]
One more thing that could be done is to only display (in Workflow#index) those workflows that the user has permissions to submit. (user is a member of a group which is allowed to submit the form)
@jainendra ?

File: forms/templates/forms/workflows/show.html [hunk 1]
```diff
@@ -13,6 +13,8 @@
         </div>
         <div class="text-muted">
             {{ workflow.description }}
+            <br>
+            Submittable by {{ workflow.allowed_groups.all|values_list:"name"|join:", "|truncatewords:7 }}
```
Comment (jainendra) [inline forms/templates/forms/workflows/show.html]: Can you  verify if it looks natural?

**Comment by jainendra** [comment]
@nishanthvijayan What do you suggest for not showing forms that are not allowed to submit? 
Should I merge this much and solve the display stuff in other PR?

**Comment by nishanthvijayan** [comment]
Will do that in another PR :+1:

**Comment by jainendra** [comment]
Done

**Pull request merged**